edge-functions/edge-config-statsig/middleware.ts

```diff
@@ -0,0 +1,47 @@
+import { NextRequest, NextResponse } from 'next/server'
+import Statsig from "statsig-node";
+import { EdgeConfigDataAdapter } from "statsig-node-vercel"
+import { EXPERIMENT, UID_COOKIE } from './lib/constants'
+
+// We'll use this to validate a random UUID
+const IS_UUID = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{16}$/i
+
+export async function middleware(req: NextRequest) {
+  // If the request is not for `/`, continue
+  if (req.nextUrl.pathname !== '/') return
+
+  // Get the user ID from the cookie or get a new one
+  let userId = req.cookies.get(UID_COOKIE)?.value
+  let hasUserId = !!userId
+
+  // If there's no active user ID in cookies or its value is invalid, get a new one
+  if (!userId || !IS_UUID.test(userId)) {
+    userId = crypto.randomUUID()
+    hasUserId = false
+  }
+
+  const dataAdapter = new EdgeConfigDataAdapter(process.env.EDGE_CONFIG_ITEM_KEY!);
```

Comment:
It would slightly improve latency if we moved the `dataAdapter` declaration outside of the middleware function so that it can be reused across requests